### ISSUE TITLE
csi-attacher: Failed to list *v1beta1.VolumeAttachment

### ISSUE THREAD

**Issue by iPenx** [cubefs/cubefs-csi#63]
Kubernetes Version: 1.22.2
csi-attacher Version: quay.io/k8scsi/csi-attacher:v2.0.0
github yaml commit: df3943c0f4f26ac28c018dde85f7419445917922 (branch master now)

**Comment by Huweicai** [comment]
Would you please provide more detailed error message?

**Comment by iPenx** [comment]
Oh, Sorry. I forgot to attach the error message.

```
$ kubectl -n cubefs logs cfs-csi-controller-6fd498dbdb-5hmhg -c csi-attacher
...
...
...
E0223 15:01:44.977981       1 reflector.go:126] k8s.io/client-go/informers/factory.go:133: Failed to list *v1beta1.VolumeAttachment: the server could not find the requested resource
E0223 15:01:45.989032       1 reflector.go:126] k8s.io/client-go/informers/factory.go:133: Failed to list *v1beta1.VolumeAttachment: the server could not find the requested resource
E0223 15:01:46.991664       1 reflector.go:126] k8s.io/client-go/informers/factory.go:133: Failed to list *v1beta1.VolumeAttachment: the server could not find the requested resource
E0223 15:01:47.997030       1 reflector.go:126] k8s.io/client-go/informers/factory.go:133: Failed to list *v1beta1.VolumeAttachment: the server could not find the requested resource
E0223 15:01:49.000336       1 reflector.go:126] k8s.io/client-go/informers/factory.go:133: Failed to list *v1beta1.VolumeAttachment: the server could not find the requested resource
E0223 15:01:50.003900       1 reflector.go:126] k8s.io/client-go/informers/factory.go:133: Failed to list *v1beta1.VolumeAttachment: the server could not find the requested resource
E0223 15:01:51.005052       1 reflector.go:126] k8s.io/client-go/informers/factory.go:133: Failed to list *v1beta1.VolumeAttachment: the server could not find the requested resource
E0223 15:01:52.006395       1 reflector.go:126] k8s.io/client-go/informers/factory.go:133: Failed to list *v1beta1.VolumeAttachment: the server could not find the requested resource
E0223 15:01:53.009911       1 reflector.go:126] k8s.io/client-go/informers/factory.go:133: Failed to list *v1beta1.VolumeAttachment: the server could not find the requested resource
E0223 15:01:54.012785       1 reflector.go:126] k8s.io/client-go/informers/factory.go:133: Failed to list *v1beta1.VolumeAttachment: the server could not find the requested resource
E0223 15:01:55.015882       1 reflector.go:126] k8s.io/client-go/informers/factory.go:133: Failed to list *v1beta1.VolumeAttachment: the server could not find the requested resource
E0223 15:01:56.032561       1 reflector.go:126] k8s.io/client-go/informers/factory.go:133: Failed to list *v1beta1.VolumeAttachment: the server could not find the requested resource
E0223 15:01:57.035013       1 reflector.go:126] k8s.io/client-go/informers/factory.go:133: Failed to list *v1beta1.VolumeAttachment: the server could not find the requested resource
E0223 15:01:58.036868       1 reflector.go:126] k8s.io/client-go/informers/factory.go:133: Failed to list *v1beta1.VolumeAttachment: the server could not find the requested resource
E0223 15:01:59.039020       1 reflector.go:126] k8s.io/client-go/informers/factory.go:133: Failed to list *v1beta1.VolumeAttachment: the server could not find the requested resource
E0223 15:02:00.041966       1 reflector.go:126] k8s.io/client-go/informers/factory.go:133: Failed to list *v1beta1.VolumeAttachment: the server could not find the requested resource
E0223 15:02:01.043263       1 reflector.go:126] k8s.io/client-go/informers/factory.go:133: Failed to list *v1beta1.VolumeAttachment: the server could not find the requested resource
E0223 15:02:02.045641       1 reflector.go:126] k8s.io/client-go/informers/factory.go:133: Failed to list *v1beta1.VolumeAttachment: the server could not find the requested resource
E0223 15:02:03.048531       1 reflector.go:126] k8s.io/client-go/informers/factory.go:133: Failed to list *v1beta1.VolumeAttachment: the server could not find the requested resource
E0223 15:02:04.051757       1 reflector.go:126] k8s.io/client-go/informers/factory.go:133: Failed to list *v1beta1.VolumeAttachment: the server could not find the requested resource
E0223 15:02:05.053018       1 reflector.go:126] k8s.io/client-go/informers/factory.go:133: Failed to list *v1beta1.VolumeAttachment: the server could not find the requested resource
E0223 15:02:06.071530       1 reflector.go:126] k8s.io/client-go/informers/factory.go:133: Failed to list *v1beta1.VolumeAttachment: the server could not find the requested resource
E0223 15:02:07.075055       1 reflector.go:126] k8s.io/client-go/informers/factory.go:133: Failed to list *v1beta1.VolumeAttachment: the server could not find the requested resource
E0223 15:02:08.080017       1 reflector.go:126] k8s.io/client-go/informers/factory.go:133: Failed to list *v1beta1.VolumeAttachment: the server could not find the requested resource
E0223 15:02:09.081374       1 reflector.go:126] k8s.io/client-go/informers/factory.go:133: Failed to list *v1beta1.VolumeAttachment: the server could not find the requested resource
E0223 15:02:10.084753       1 reflector.go:126] k8s.io/client-go/informers/factory.go:133: Failed to list *v1beta1.VolumeAttachment: the server could not find the requested resource
E0223 15:02:11.085884       1 reflector.go:126] k8s.io/client-go/informers/factory.go:133: Failed to list *v1beta1.VolumeAttachment: the server could not find the requested resource
E0223 15:02:12.089213       1 reflector.go:126] k8s.io/client-go/informers/factory.go:133: Failed to list *v1beta1.VolumeAttachment: the server could not find the requested resource
```

**Comment by Huweicai** [comment]
Sorry we haven't tested our csi in kubernetes relase v1.22, it seems some api is deprecated, we need some time to support it, appreciate for your reporting.

https://kubernetes.io/docs/reference/using-api/deprecation-guide/#storage-resources-v122
> The storage.k8s.io/v1beta1 API version of CSIDriver, CSINode, StorageClass, and VolumeAttachment is no longer served as of v1.22.
Migrate manifests and API clients to use the storage.k8s.io/v1 API version

**Comment by iPenx** [comment]
ok, I see. Thank you.

**Comment by fei922** [comment]
csi-attacher image tag upgrade to support k8s version 1.22 or above
![image](https://user-images.githubusercontent.com/9672023/187115582-619825c6-c311-4c0d-be73-c052cb3f036a.png)

**Comment by Huweicai** [comment]
3.1.0.150.0 was released, kubernetes v1.22 or later versions are supported now.
